pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,160 @@ type ExternalNodeList struct {
+)
+
+type SupportBundleCollectionStatus struct {
+	Phase BundlePhase `json:"phase"`
+	// The generation observed by Antrea.
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// The number of Nodes and ExternalNodes that have completed the SupportBundleCollection.
+	CollectedNodes int32 `json:"collectedNodes"`
+	// The total number of Nodes and ExternalNodes that should process the SupportBundleCollection.
+	DesiredNodes int32                              `json:"desiredNodes"`
+	Conditions   []SupportBundleCollectionCondition `json:"conditions"`
+}
+
+type SupportBundleCollectionConditionType string
+
+const (
+	// Started means Antrea Controller has created the internal resource for the SupportBundleCollectionCondition.
+	Started SupportBundleCollectionConditionType = "started"
+	// Collected means all Antrea Agents have successfully processed the SupportBundleCollectionCondition request.
+	Collected SupportBundleCollectionConditionType = "collected"
+)
+
+// SupportBundleCollectionCondition describes the state of a SupportBundleCollection at a certain point.
+type SupportBundleCollectionCondition struct {
+	// Type of StatefulSet condition.
+	Type SupportBundleCollectionConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status metav1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// The reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// A human-readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+type BundleNodes struct {
+	// List the names of certain Nodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain Nodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type BundleExternalNodes struct {
+	Namespace string
+	// List the names of certain ExternalNodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain ExternalNodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+// BundleUploadProtocol defines the scheme that the BundleFileServer uses in the url to
+// upload bundle files.
+type BundleUploadProtocol string
+
+const (
+	HTTPS BundleUploadProtocol = "https"
+)
+
+// BundleUploadVerb defines the verb to upload bundle files to the URL provided by
+// the BundleFileServer.
+type BundleUploadVerb string
+
+const (
+	PostBundle BundleUploadVerb = "post"
+)
+
+// BundleFileServer specifies the bundle file server information.
+type BundleFileServer struct {
+	HTTPServer string               `json:"httpServer"`
```

Comment:
I thought over. I feel we should either use your original way of "host" + "path" + "port", or change to URL (which should be like https://host:port/path). Which way you like?

If we go the former way, I think we should define a struct for the three fields.